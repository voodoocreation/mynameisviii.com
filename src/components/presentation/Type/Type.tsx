import cn from "classnames";
import * as React from "react";
import { MdAlbum, MdEvent, MdNewReleases } from "react-icons/md";
import { InjectedIntl, injectIntl } from "react-intl";

interface IProps {
  className?: string;
  hasLabel?: boolean;
  intl: InjectedIntl;
  value: string;
}

const Type: React.SFC<IProps> = ({
  className,
  hasLabel,
  intl,
  value,
  ...props
}) => {
  let icon;
  let label;

  switch (value) {
    case "release":
      icon = <MdAlbum />;
      label = intl.formatMessage({ id: "RELEASE" });
      break;

    case "appearance":
      icon = <MdEvent />;
      label = intl.formatMessage({ id: "APPEARANCE" });
      break;

    default:
      icon = <MdNewReleases />;
      label = intl.formatMessage({ id: "NEWS" });
      break;
  }

  return (
    <div
      className={cn("Type", `Type-${value}`, className)}
      title={!hasLabel ? label : undefined}
      {...props}
    >
      {icon}
      {hasLabel ? ` ${label}` : null}
    </div>
  ) as React.ReactElement<any>;
};

Type.defaultProps = {
  hasLabel: true
};

export default injectIntl<any>(Type);
