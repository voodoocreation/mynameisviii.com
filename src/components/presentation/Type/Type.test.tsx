import { render } from "enzyme";
import * as React from "react";

import Type from "./Type";

const setup = (fn: any, fromTestProps?: any) => {
  const props = {
    className: "TestType",
    value: "news",
    ...fromTestProps
  };

  return {
    actual: fn(<Type {...props} />),
    props
  };
};

describe("[presentation] <Type />", () => {
  it("renders correctly with minimum props", () => {
    const { actual } = setup(render);

    expect(actual).toMatchSnapshot();
  });

  it("renders correctly without label", () => {
    const { actual } = setup(render, { hasLabel: false });

    expect(actual).toMatchSnapshot();
  });

  it("renders correctly when value=release", () => {
    const { actual } = setup(render, { value: "release" });
    expect(actual).toMatchSnapshot();
  });

  it("renders correctly when value=appearance", () => {
    const { actual } = setup(render, { value: "appearance" });

    expect(actual).toMatchSnapshot();
  });
});
