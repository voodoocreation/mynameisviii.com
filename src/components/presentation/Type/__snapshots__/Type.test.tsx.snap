// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[presentation] <Type /> renders correctly when value=appearance 1`] = `
<div
  class="Type Type-appearance TestType"
>
  <svg
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style="vertical-align:middle"
    viewBox="0 0 40 40"
    width="1em"
  >
    <g>
      <path
        d="m31.6 31.6v-18.2h-23.2v18.2h23.2z m-5-30h3.4v3.4h1.6q1.4 0 2.4 1t1 2.4v23.2q0 1.4-1 2.4t-2.4 1h-23.2q-1.4 0-2.4-1t-1-2.4v-23.2q0-1.4 1-2.4t2.4-1h1.6v-3.4h3.4v3.4h13.2v-3.4z m1.8 18.4v8.4h-8.4v-8.4h8.4z"
      />
    </g>
  </svg>
   Appearance
</div>
`;

exports[`[presentation] <Type /> renders correctly when value=release 1`] = `
<div
  class="Type Type-release TestType"
>
  <svg
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style="vertical-align:middle"
    viewBox="0 0 40 40"
    width="1em"
  >
    <g>
      <path
        d="m20 18.4c0.9 0 1.6 0.7 1.6 1.6s-0.7 1.6-1.6 1.6-1.6-0.7-1.6-1.6 0.7-1.6 1.6-1.6z m0 9.1c4.1 0 7.5-3.4 7.5-7.5s-3.4-7.5-7.5-7.5-7.5 3.4-7.5 7.5 3.4 7.5 7.5 7.5z m0-24.1c9.2 0 16.6 7.4 16.6 16.6s-7.4 16.6-16.6 16.6-16.6-7.4-16.6-16.6 7.4-16.6 16.6-16.6z"
      />
    </g>
  </svg>
   Release
</div>
`;

exports[`[presentation] <Type /> renders correctly with minimum props 1`] = `
<div
  class="Type Type-news TestType"
>
  <svg
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style="vertical-align:middle"
    viewBox="0 0 40 40"
    width="1em"
  >
    <g>
      <path
        d="m21.6 21.6v-10h-3.2v10h3.2z m0 6.8v-3.4h-3.2v3.4h3.2z m16.8-8.4l-4.1 4.6 0.5 6.2-6 1.3-3.1 5.3-5.7-2.4-5.7 2.4-3.1-5.2-6-1.4 0.5-6.2-4.1-4.6 4.1-4.7-0.5-6.1 6-1.3 3.1-5.3 5.7 2.4 5.7-2.4 3.1 5.3 6 1.3-0.5 6.2z"
      />
    </g>
  </svg>
   News
</div>
`;

exports[`[presentation] <Type /> renders correctly without label 1`] = `
<div
  class="Type Type-news TestType"
  title="News"
>
  <svg
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style="vertical-align:middle"
    viewBox="0 0 40 40"
    width="1em"
  >
    <g>
      <path
        d="m21.6 21.6v-10h-3.2v10h3.2z m0 6.8v-3.4h-3.2v3.4h3.2z m16.8-8.4l-4.1 4.6 0.5 6.2-6 1.3-3.1 5.3-5.7-2.4-5.7 2.4-3.1-5.2-6-1.4 0.5-6.2-4.1-4.6 4.1-4.7-0.5-6.1 6-1.3 3.1-5.3 5.7 2.4 5.7-2.4 3.1 5.3 6 1.3-0.5 6.2z"
      />
    </g>
  </svg>
</div>
`;
